File: interlayer/src/main/java/com/street/interlayer/model/Wallet.java
package com.street.interlayer.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
public class Wallet {

    private String id;
    @JsonProperty("owner")
    private Long ownerId;
    private Account account;
    private Double amount;
    private List<Transaction> transactions;

}

File: micro/src/main/kotlin/com/street/micro/entity/Wallet.kt
package com.street.micro.entity

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field

@Document
data class Wallet(
    @Id
    @JsonSerialize(using = ToStringSerializer::class)
    val id: ObjectId = ObjectId.get(),
    val owner: Long,
    var amount: Double,
    @Field("transaction_ids")
    val transactionIds: List<ObjectId>,
    var transactions: List<Transaction>?
)